src/components/storage/aws.ts

```diff
@@ -24,41 +20,26 @@ export class AwsUserStorage implements UserStorage {
         this.bucketName = bucketName
       }
 
-    async storeUser(id:string, install_time: number) {
-        const key = 'users/users.csv'
-
-        const users = await this._getObject({key, type: 'csv'})
-        users.push([id, install_time])
-
-        let body;
-
-        await new Promise((resolve, reject) => {
-            stringify(users, function(err, output){
-                body = output;
-                resolve(output)
-            });
-        })
+    async storeUser(id:string, installTime: number) {
+        const key = 'users/' + id + '.json'
```

Comment:
There's lots of these string and object literals around this class. They should be set as properties on the class instance, so you could go `const key = this.dir + id + this.ext` (names are just a suggestion). 
Think about if you have to change any of these and then you need to search through your code to find all occurrences. Generally you would want them to be able to be set from the `constructor`, so external modules can set these. But you may not want to specify them every time you create an instance, so you could set their default values to be static class properties in the constructor args. 

tl;dr: use class props to DRY out all these literal values

Quick example (just a suggestion; add things as appropriate):

```ts
class Test {
  static DEF_DIR = 'users/'
  static DEF_EXT = '.json'
  static DEF_CONTENT_TYPE = {json: 'text/json', ...}

  constructor({bucketName, dir = Test.DEF_DIR, ext = Test.DEF_EXT, contentType = Test.DEF_CONTENT_TYPE}) {
    this.dir = dir
    this.ext = ext
    ... // init everything else
  }
}
```

On the topic of class props:
Instance props `_s3` and `bucketName` are both `public` but I don't see them being accessed outside the class methods. I would opt for making your instance props `private` by default unless you have a good reason not to (you can also make use of setters/getters for outside access. Good info: https://www.typescriptlang.org/docs/handbook/classes.html#accessors).